generateFiles.py

```diff
@@ -0,0 +1,147 @@
+import re
+import sys
+import subprocess
+import os
+import rtoml
+
+# Environment variables
+github_repository = os.environ.get('GITHUB_REPOSITORY').split('/')
+idf_version = os.environ.get('IDF_VERSION')
+
+github_owner = github_repository[0]
+github_repo = github_repository[1]
+
+# Root toml object
+toml_obj = {'esp_toml_version': 1.0, 'firmware_images_url': f'https://{github_owner}.github.io/{github_repo}/', 'supported_apps': []}
+
+class App:
+    
+    def __init__(self, app):
+        # App directory
+        self.app_dir = app
+        # Name of the app
+        if app:
+            self.name = app.split('/')[-1]
+        # List of targets (esp32, esp32s2, etc)
+        self.targets = []
+        # List of tuples (kit, target)
+        self.boards = []
+
+current_app = App(None)
+
+# Regex to get the app_dir
+def get_app_dir(line):
+    return re.search(r'"app_dir":\s*"([^"]*)",', line).group(1) if re.search(r'"app_dir":\s*"([^"]*)",', line) else None
+
+# Regex to get the target
+def get_target(line):
+    return re.search(r'"target":\s*"([^"]*)",', line).group(1) if re.search(r'"target":\s*"([^"]*)",', line) else None
+
+# Regex to get the kit
+def get_kit(line):
+    return re.search(r'"config":\s*"([^"]*)",', line).group(1) if re.search(r'"config":\s*"([^"]*)",', line) else None
+
+# Squash the json into a list of apps
+def squash_json(input_str):
+    global current_app
+    # Split the input into lines
+    lines = input_str.splitlines()
+    output_list = []
+    for line in lines:
+        # Get the app_dir
+        app = get_app_dir(line)
+        # If its a not a None and not the same as the current app
+        if current_app.app_dir != app:
+            # Save the previous app
+            if current_app.app_dir:
+                output_list.append(current_app.__dict__)
+            current_app = App(app)
+
+        # If we are building for a kit        
+        if (get_kit(line) != ''):
+            current_app.boards.append((get_kit(line), get_target(line)))
+        # If we are building for targets
+        else:
+            current_app.targets.append(get_target(line))
+
+    # Append last app
+    output_list.append(current_app.__dict__)
+    
+    return output_list
+
+# Merge binaries for each app
+def merge_binaries(apps):
+    subprocess.run(['mkdir', 'binaries'])
+    for app in apps:
+        # If we are merging binaries for kits
+        if app.get('boards'):
+            for board in app['boards']:
+                kit = board[0]
+                target = board[1]
+                cmd = ['esptool.py', '--chip', target, 'merge_bin', '-o', app['name'] + '-' + kit + '-' + target + '-' + idf_version + '.bin', '@flash_args']
```

Comment:
nitpick: you are joining strings here and using f-strings later (`f'{app["name"]}-{kit}-{target}-{idf_version}.bin'`). I would recommend using f-strings everywhere.

Reply:
@radimkarnis 

Thanks you for taking a look, I fixed it!